app/models/manageiq/providers/google/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,164 @@
+        _log.warn("Received multiple values for cpu_usage; ignoring duplicates")
+        x
+      end
+    },
+    {
+      :vim_counter_name        => "disk_usage_rate_average",
+      :google_metric_names     => ["compute.googleapis.com/instance/disk/read_bytes_count",
+                                   "compute.googleapis.com/instance/disk/write_bytes_count"],
+      :target_to_google_labels => ->(target) { ["compute.googleapis.com/resource_id==#{target.ems_ref}"] },
+      :point_to_val            => ->(point) { point["int64Value"].to_i / (60.0 * 1024.0) }, # convert from b/m to Kb/s
+      :reducer                 => ->(x, y) { x + y },
+    },
+    {
+      :vim_counter_name        => "net_usage_rate_average",
+      :google_metric_names     => ["compute.googleapis.com/instance/network/received_bytes_count",
+                                   "compute.googleapis.com/instance/network/sent_bytes_count"],
+      :target_to_google_labels => ->(target) { ["compute.googleapis.com/resource_id==#{target.ems_ref}"] },
+      :point_to_val            => ->(point) { point["int64Value"].to_i / (60.0 * 1024.0) }, # convert from b/m to Kb/s
+      :reducer                 => ->(x, y) { x + y },
+    }
+  ].freeze
+
+  def perf_collect_metrics(_interval_name, start_time = nil, end_time = nil)
+    raise "No EMS defined" if target.ext_management_system.nil?
+
+    # Currently we only know how to capture VMs
+    return [{}, {}] unless target.type == ManageIQ::Providers::Google::CloudManager::Vm.name
+
+    end_time   ||= Time.now.utc
+    end_time     = end_time.utc
+    start_time ||= end_time - 4.hours # 4 hours for symmetry with VIM (still needed?)
+    start_time   = start_time.utc
+
+    counter_values_by_ts = {}
+    VIM_COUNTER_SCHEMAS.each do |schema|
+      collect_metrics(start_time, end_time, schema, counter_values_by_ts)
+    end
+
+    counters_by_id              = {target.ems_ref => VIM_STYLE_COUNTERS}
+    counter_values_by_id_and_ts = {target.ems_ref => counter_values_by_ts}
+
+    return counters_by_id, counter_values_by_id_and_ts
+  end
+
+  # Lookup and retrieve a metric from Google Cloud, storing it in the
+  # 'counter_values_by_ts' hash. Note that this method retrieves, transforms
+  # and collects a metric according to a metric schema (see
+  # VIM_COUNTER_SCHEMAS for examples).
+  #
+  # @param start_time [Time] the earliest point to query
+  # @param end_time [Time] the latest point to query
+  # @param schema [Hash] schema describing metric to query (see
+  #   VIM_STYLE_COUNTERS definition for a description)
+  # @param counter_values_by_ts [Hash{Time => Hash{String => Number}}] hash to
+  #   aggregate metrics onto (will be modified by method)
+  # @return nil
+  def collect_metrics(start_time, end_time, schema, counter_values_by_ts)
+    schema[:google_metric_names].each do |google_metric_name|
+      target.ext_management_system.with_provider_connection(:service => 'monitoring') do |google|
```

Comment:
Can you connect in `#perf_collect_metrics` so you only open the connection once and use it for all the metrics types?